target/linux/ramips/image/mt76x8.mk

```diff
@@ -6,6 +6,26 @@ include ./common-tp-link.mk
 
 DEFAULT_SOC := mt7628an
 
+define Build/creality_wb-01-factory
+  echo '#!/bin/sh' > $@.install.sh
+  echo 'kernel_size=851968' >> $@.install.sh
+  echo 'file=$$2/factory.bin' >> $@.install.sh
+  echo 'file_size=$$(wc -c < $$file)' >> $@.install.sh
+  echo 'fs_size=$$((file_size - kernel_size))' >> $@.install.sh
+  echo 'mtd_write -o 0 -l $$kernel_size write $$file Kernel' \
+    >> $@.install.sh
+  echo 'mtd_write -r -o $$kernel_size -l $$fs_size write $$file RootFS' \
+    >> $@.install.sh
```

Comment:
I wonder whether this can be written at once with some <-EOF syntax ...

Reply:
I don't think so? It seems as though you have to use `.ONESHELL:` to do heredoc syntax in make but that caused errors like `/usr/bin/env bash: No such file or directory` for me when I tried, and it might also cause problems somewhere else because of how that directive works. There might be a better way but I can't think of one.